### DISCUSSION TITLE
Graph neural network about member inference attacks

### DISCUSSION THREAD

**Discussion by Chrisable** [pyg-team/pytorch_geometric#8792] [General]
GCN is transductive learning, how to judge whether a node on a graph is a member? [This paper](https://arxiv.org/pdf/2101.06570.pdf) provides a method of judgment in inductive learning，But,transductive learning?

**Comment by ArnabKumarRoy02** [comment]
Yes, `transductive learning` concentrates on generating predictions for the nodes that are already present in the graph, as opposed to `inductive learning`, where the model must generalise to unknown nodes.

For judging whether a node is a member of the graph, you can:
 - Use a `trained model` to predict the labels of that particular node.
 - Set a `threshold` for prediction (just like confidence of that node being a member), `pred <= threshold` means not a member.
 - `Divide` the graph into several folds, train on a `portion` of it, and then evaluate your performance using the `remaining data`.
 - `Modifying` hyper-parameters like `learning rate` and `layer number`, the model's performance on the transductive learning task can be `optimised`.

These are not different methods, these are the steps to follow to solve you question.

**Reply by Chrisable** [reply]
It's okay to use this [paper](https://arxiv.org/pdf/2101.06570.pdf) directly

**Reply by Chrisable** [reply]
Looking forward to  your reply!

**Reply by ArnabKumarRoy02** [reply]
Yes, it's okay to use this paper directly. As the paper mentions that GNN models could `leak information` about the `member nodes` that were trained on. 
So, if your primary question is to check whether a node on a graph is a member or not? You `CAN` use this paper.
